From: Sascha Hauer <s.hauer@pengutronix.de>
Date: Fri, 8 Aug 2014 14:47:18 +0200
Subject: [PATCH] xf86-video-fbdev: add option to leave timings alone

Xorg tries to set the timing values in the framebuffer device. Add an
option to tell X to leave the timing values untouched.

Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
---
 src/fbdev.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/fbdev.c b/src/fbdev.c
index f25ef72204f0..35a3c60f7835 100644
--- a/src/fbdev.c
+++ b/src/fbdev.c
@@ -125,6 +125,7 @@ typedef enum {
 	OPTION_SHADOW_FB,
 	OPTION_ROTATE,
 	OPTION_FBDEV,
+	OPTION_DONT_SET_TIMING,
 	OPTION_DEBUG
 } FBDevOpts;
 
@@ -132,6 +133,7 @@ static const OptionInfoRec FBDevOptions[] = {
 	{ OPTION_SHADOW_FB,	"ShadowFB",	OPTV_BOOLEAN,	{0},	FALSE },
 	{ OPTION_ROTATE,	"Rotate",	OPTV_STRING,	{0},	FALSE },
 	{ OPTION_FBDEV,		"fbdev",	OPTV_STRING,	{0},	FALSE },
+	{ OPTION_DONT_SET_TIMING, "DontSetTiming", OPTV_BOOLEAN, {0},	FALSE },
 	{ OPTION_DEBUG,		"debug",	OPTV_BOOLEAN,	{0},	FALSE },
 	{ -1,			NULL,		OPTV_NONE,	{0},	FALSE }
 };
@@ -589,9 +591,11 @@ FBDevPreInit(ScrnInfoPtr pScrn, int flags)
 	}
 
 	/* select video modes */
-
-	xf86DrvMsg(pScrn->scrnIndex, X_INFO, "checking modes against framebuffer device...\n");
-	fbdevHWSetVideoModes(pScrn);
+	if( xf86ReturnOptValBool(fPtr->Options, OPTION_DONT_SET_TIMING, FALSE) == FALSE ) {
+		xf86DrvMsg(pScrn->scrnIndex, X_INFO, "checking modes against framebuffer device...\n");
+		fbdevHWSetVideoModes(pScrn);
+	} else
+		xf86DrvMsg(pScrn->scrnIndex, X_INFO, "ignoring fb timing settings.");
 
 	xf86DrvMsg(pScrn->scrnIndex, X_INFO, "checking modes against monitor...\n");
 	{
